analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -90,11 +90,20 @@ private IEnumerable<ExplodedNode> ProcessBranching(ExplodedNode node)
         {
             if (node.Block.Kind == BasicBlockKind.Exit)
             {
-                checks.ExitReached(new(null, node.State));
+                NotifyExitReached();
             }
-            else if (node.Block.ContainsThrow())
+            else if (node.Block.ContainsThrowOrRethrow())
             {
-                yield return new(cfg.ExitBlock, node.State, null);
+                var successors = ExceptionSuccessors(node, ThrowExceptionType(node.Block.BranchValue)).ToList();
+                foreach (var successor in successors)
+                {
+                    yield return successor;
+                }
+
+                if (successors.Count == 0)
+                {
+                    NotifyExitReached();
+                }
```

Comment:
I've found the problem. This should not be here. Instead, on re-throw, we should not call `ThrowExceptionType`, but reuse the previous `state.Exception`, if not null.

Reply:
I didn't try, let's see if the modification will break some UT. If not, we'll have to craft a special case for this.

Reply:
If it's `null` and `rethrow`, we don't know where to continue => `yield break;` (or just leave a comment and let the method exit)